standard-cluster-services/src/main/java/org/terracotta/monitoring/PlatformService.java

```diff
@@ -54,4 +54,51 @@
    * Provides the raw config file from server startup
    */
   InputStream getPlatformConfiguration();
+
+  /**
+   * Stops the server if it is in any of the passive states
+   * <p>
+   * Based on the {@code restartMode} argument, the server should either bounce or just halt (exit with correct exit codes).
+   * Currently, it is assumed that the halt may be abrupt and it may not clean up or empty any queues.
+   *
+   * @param restartMode mode in which the system needs to restart. STOP_ONLY means halt without restarting
+   */
+  void stopPlatformIfPassive(RestartMode restartMode);
+
+  /**
+   * Stops the server if it is in any of the active states (including active suspended)
+   * <p>
+   * Based on the {@code restartMode} argument, the server should either bounce or just halt.
+   * Currently, it is assumed that the halt may be abrupt and it may not clean up or empty any queues.
+   *
+   * @param restartMode mode in which the system needs to restart. STOP_ONLY means halt without restarting
+   */
+  void stopPlatformIfActive(RestartMode restartMode);
+
+  /**
+   * Stops the server.
+   * <p>
+   * Based on the {@code restartMode} argument, the server should either bounce or just halt (exit with correct exit codes).
+   * Currently, it is assumed that the halt may be abrupt and it may not clean up or empty any queues.
+   *
+   * @param restartMode mode in which the system needs to restart. STOP_ONLY means halt without restarting
+   */
+  void stopPlatform(RestartMode restartMode);
+
+  /**
+   * Returns true if it is safe to stop.
+   * <p>
+   * {@code partialStop} is true means this is a rolling stop (for rolling upgrades etc). Otherwise, this is a full stripe shutdown
+   * <p>
+   * If partialStop is true and if this server is ACTIVE_COORDINATOR, then return true iff all other stripe members are in PASSIVE_STANDBY, false otherwise
+   * If partialStop is true and if this server is PASSIVE_STANDBY, then return true
+   * If partialStop is true and if this server is any other state, then return false
+   * If partialStop is false, return true always (for now)
+   *
+   */
+  boolean isSafeToStop(boolean partialStop);
+
+  enum  RestartMode {
+    STOP_ONLY, STOP_AND_RESTART_IN_SAFE_MODE, STOP_AND_RESTART
```

Comment:
SAFE_MODE doesn't mean anything when the stripe is running and should be removed.

Reply:
this is for restarting/bouncing when an entire cluster is asked to stop and restart, 

@albinsuresh if you think this is not needed anymore (not sure about the outcome of all the restart discussions) even for AVAILABILITY failover mode, please let me know. We can remove this.

Reply:
@ramsai1729, please remove the STOP_AND_RESTART_IN_SAFE_MODE from above.

Reply:
done

Reply:
We still have safe mode startup in availability mode.

Reply:
added STOP_AND_RESTART_IN_SAFE_MODE back

Reply:
I don't see any reasonable use-case scenario for this mode.  Please remove it and I will merge.